linera-service/src/server.rs

```diff
@@ -180,8 +182,15 @@ where
     type Output = ();
 
     async fn run(self, storage: S) -> Result<(), anyhow::Error> {
-        // Allow local IP address to be different from the public one.
-        let listen_address = "0.0.0.0";
+        let tmp_listen_address = if !self.kube {
+            // Allow local IP address to be different from the public one.
+            "0.0.0.0".to_string()
+        } else {
+            std::env::var("MY_POD_IP").expect("Could not get env variable MY_POD_IP")
+        };
+
+        let listen_address = tmp_listen_address.as_str();
```

Comment:
Looks like `spawn_simple` and `spawn_grpc` both turn the `&str` back into a `String`, so we might as well pass in the `String`.

Reply:
That's a good point, but if I use `String` there's an `async move` somewhere down that messes everything up 😩
Leaving as is now, but if there's an easy way to get around that that I'm missing, please lmk

Reply:
@andresilva91 We generally don't define aliases for references. Just write `let listen_address = if !self.kube { ... } else { ... }` then use `&listen_address` (or `listen_address.clone()`) afterwards when the compiler requires it.

Reply:
Sounds good!